shared/src/main/scala/mlscript/Typer.scala

```diff
@@ -81,6 +123,18 @@ class Typer(var dbg: Boolean, var verbose: Bool, var explainErrors: Bool, val ne
     private def containsMth(key: (Str, Str) \/ (Opt[Str], Str)): Bool = mthEnv.contains(key) || parent.exists(_.containsMth(key))
     def containsMth(parent: Opt[Str], nme: Str): Bool = containsMth(R(parent, nme))
     def nest: Ctx = copy(Some(this), MutMap.empty, MutMap.empty)
+    /**
+      * Enter a new quoted environment with empty `quoteSkolemEnv` and `freeVarsInCurrentQuote`.
+      * A whole quasiquote (i.e., code"...") contains no binding or free variables at the beginning.
+      * For a quoted binding (e.g., code"x => ..."):
+      *  1. The context of the lambda body is still in the quotation so `inQuote = true`
+      *  2. `quoteSkolemEnv` only contains skolems created by the current binding and `freeVarsInCurrentQuote` only contains free variables appearing in the body.
+      * So we also apply empty `quoteSkolemEnv` and `freeVarsInCurrentQuote` at the beginning.
+      * e.g. `code"x => y => x + y"`. For `y => x + y`, freeVarsInCurrentQuote = {\ga_x, \ga_y}, quoteSkolemEnv = {\ga_y}. 
+      * So for `code"x => ..."`, freeVarsInCurrentQuote = {\al}, quoteSkolemEnv = {\ga_x}, where \ga_x \/ \ga_y <= \al \/ \ga_y
```

Comment:
What is `\al` here? It seems to come out of nowhere, as it did not appear on the previous line.